gym_cap/envs/create_map.py

```diff
@@ -85,46 +85,65 @@ def gen_map(name, dim=20, in_seed=None, rand_zones=False, np_random=None,
         return new_map, static_map
     
     @staticmethod
-    def dest_map(new_map,\
-                 map_obj=[NUM_BLUE, NUM_UAV, NUM_RED, NUM_UAV, NUM_GRAY]):
+    def dest_map(new_map):
         """
         Method
-            Outputs static_map when new_map is given as input
-            Addtionally the number of agents can also be changed
+            Outputs static_map when new_map is given as input.
+            Addtionally the number of agents will also be
+            counted
         
         Parameters
         ----------
-        agent_map        : numpy array
+        new_map        : numpy array
             new_map
-        map_obj     : list
-            The necessary elements to build the map
-            0   : blue UGV
-            1   : blue UAV
-            2   : red UGV
-            3   : red UAV
-            4   : gray units
+        The necessary elements:
+            ugv_1   : blue UGV
+            uav_1   : blue UAV
+            ugv_2   : red UGV
+            uav_2   : red UAV
+            gray    : gray units
+            
         """
         orig_map = np.copy(new_map)
-        for j in range(map_obj[0]):
+        ugv_1 = 0
+        ugv_2 = 0
+        uav_1 = 0
+        uav_2 = 0
+        gray = 0
+        for i in range(0, new_map.shape[0]):
+            for j in range(0, new_map.shape[1]):
+                if map[i][j] == TEAM1_UGV:  # number of team 1 UGV
+                    ugv_1 += 1
+                elif map[i][j] == TEAM2_UGV:  # number of team 2 UGV
+                    ugv_2 += 1
+                elif map[i][j] == TEAM1_UAV:  # number of team 1 UAV
+                    uav_1 += 1
+                elif map[i][j] == TEAM2_UAV: # number of team 2 UAV
+                    uav_2 += 1
+                elif map[i][j] == TEAM3_UGV: # number of gray agents
+                    gray += 1
+                    
+        for k in range(ugv_1):
             new_map = CreateMap.depopulate_map(new_map,
                                  TEAM1_BACKGROUND, TEAM1_UGV)
             
-        for j in range(map_obj[1]):
+        for k in range(uav_1):
             new_map = CreateMap.depopulate_map(new_map,
                                  TEAM1_BACKGROUND, TEAM1_UAV)
             
-        for j in range(map_obj[2]):
+        for k in range(ugv_2):
             new_map = CreateMap.depopulate_map(new_map,
                                  TEAM2_BACKGROUND, TEAM2_UGV)
             
-        for j in range(map_obj[3]):
+        for k in range(uav_2):
             new_map = CreateMap.depopulate_map(new_map,
                                  TEAM2_BACKGROUND, TEAM2_UAV)
 
         # TODO: change zone for grey team to complete map
-        for j in range(map_obj[4]):
+        for k in range(gray):
             new_map = CreateMap.depopulate_map(new_map,
                                  TEAM2_BACKGROUND, TEAM3_UGV)
+            
         # new_map becomes static_map    
         return orig_map, new_map
```

Comment:
Looks like orig_map is the static and new_map contains moving elements.

Reply:
new_map becomes static_map because the (moving) agents are removed from new_map in every iteration. orig_map is a copy of new_map before new_map was changed. Therefore, orig_map is the map with moving elements and new_map becomes static_map